src/handlers/spec-rewriter.ts

```diff
@@ -0,0 +1,155 @@
+import { Context } from "../types";
+import { CallbackResult } from "../types/proxy";
+import { createSpecRewriteSysMsg, llmQuery } from "./prompt";
+import { encode } from "gpt-tokenizer";
+import { RequestError } from "@octokit/request-error";
+
+export type TokenLimits = {
+  modelMaxTokenLimit: number;
+  maxCompletionTokens: number;
+  tokensRemaining: number;
+};
+
+export class SpecificationRewriter {
+  protected readonly context: Context;
+
+  constructor(context: Context) {
+    this.context = context;
+  }
+
+  async performSpecRewrite(): Promise<CallbackResult> {
+    if (this._isIssueCommentEvent(this.context)) {
+      if (this.context.payload.comment.body.trim().startsWith("/rewrite")) {
```

Comment:
```suggestion
      if (!this.context.payload.comment.body.trim().startsWith("/rewrite")) {
```

Please test your code before asking for reviews.

Reply:
also the ifs can be merged